Makefile

```diff
@@ -60,3 +60,13 @@ test-suite test-tox:
 	PATH="${PWD}/venv/bin:${PATH}" tox -e $(TOX_ENVLIST) -- $(TOX_ARGS)
 
 test-all: test-unit test-integration
+
+pip-compile: venv/bin/pip-compile
+	venv/bin/pip-compile --allow-unsafe --generate-hashes --output-file=requirements.txt requirements.in
```

Comment:
It would be nice to add a comment as to why we need to use `--allow-unsafe` here. From what I can tell, it's because we have a dependency on `setuptools` listed in `requirements.in`. For my own understanding, why is that needed?

Reply:
Added. It is needed because we use `pkg_resources` in the pip code, and `setuptools` is what provides that package